dpe/src/commands/derive_context.rs

```diff
@@ -704,30 +798,87 @@ mod tests {
             crypto: OpensslCrypto::new(),
             platform: DefaultPlatform,
         };
-        let mut dpe = DpeInstance::new(&mut env, Support::AUTO_INIT).unwrap();
+        let mut dpe = DpeInstance::new(
+            &mut env,
+            Support::AUTO_INIT | Support::RETAIN_PARENT_CONTEXT,
+        )
+        .unwrap();
 
-        DeriveChildCmd {
+        DeriveContextCmd {
             handle: ContextHandle::default(),
             data: [0; DPE_PROFILE.get_tci_size()],
-            flags: DeriveChildFlags::RETAIN_PARENT
-                | DeriveChildFlags::MAKE_DEFAULT
-                | DeriveChildFlags::CHANGE_LOCALITY,
-            tci_type: 7,
+            flags: DeriveContextFlags::RETAIN_PARENT_CONTEXT
+                | DeriveContextFlags::MAKE_DEFAULT
+                | DeriveContextFlags::CHANGE_LOCALITY,
+            tci_type: 0,
             target_locality: TEST_LOCALITIES[1],
         }
         .execute(&mut dpe, &mut env, TEST_LOCALITIES[0])
         .unwrap();
 
         assert_eq!(
-            DeriveChildCmd {
+            DeriveContextCmd {
                 handle: ContextHandle::default(),
                 data: [0; DPE_PROFILE.get_tci_size()],
-                flags: DeriveChildFlags::RETAIN_PARENT | DeriveChildFlags::CHANGE_LOCALITY,
-                tci_type: 7,
+                flags: DeriveContextFlags::RETAIN_PARENT_CONTEXT
+                    | DeriveContextFlags::CHANGE_LOCALITY,
+                tci_type: 0,
                 target_locality: TEST_LOCALITIES[1],
             }
             .execute(&mut dpe, &mut env, TEST_LOCALITIES[0]),
             Err(DpeErrorCode::InvalidArgument)
         );
     }
+
+    #[test]
+    fn test_recursive() {
+        let mut env = DpeEnv::<TestTypes> {
+            crypto: OpensslCrypto::new(),
+            platform: DefaultPlatform,
+        };
+        let mut dpe = DpeInstance::new(
+            &mut env,
+            Support::AUTO_INIT
+                | Support::RECURSIVE
+                | Support::INTERNAL_DICE
+                | Support::INTERNAL_INFO,
+        )
+        .unwrap();
+
+        // Make sure the parent handle is non-sense when not retaining.
+        assert_eq!(
+            Ok(Response::DeriveContext(DeriveContextResp {
+                handle: ContextHandle::default(),
+                parent_handle: ContextHandle([0xff; ContextHandle::SIZE]),
+                resp_hdr: ResponseHdr::new(DpeErrorCode::NoError),
+            })),
+            DeriveContextCmd {
```

Comment:
I think it would be worth calling twice in the test. In the current implementation, I think every recursive call will blow away the previous measurement.

Reply:
Done